src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/RazorSourceGenerator.Helpers.cs

```diff
@@ -107,7 +106,6 @@ private static SourceGeneratorProjectEngine GetGenerationProjectEngine(
                     options.SuppressMetadataSourceChecksumAttributes = !razorSourceGeneratorOptions.GenerateMetadataSourceChecksumAttributes;
                     options.SupportLocalizedComponentNames = razorSourceGeneratorOptions.SupportLocalizedComponentNames;
                     options.SuppressUniqueIds = razorSourceGeneratorOptions.TestSuppressUniqueIds;
-                    options.SuppressAddComponentParameter = !isAddComponentParameterAvailable;
```

Comment:
Instead of removing this line, can it become this?

```C#
options.SuppressAddComponentParameter = razorSourceGeneratorOptions.Configuration.SuppressAddComponentParameter;
```

Failing to set this option will break code gen.

Reply:
AFIACT this is the wanted design though based on https://github.com/dotnet/razor/pull/10763#discussion_r1731887010 

It's very possible I missed switching something to check the configuration (or rather @chsienki 😜 )

Reply:
The option still needs to be set on the RazorCodeGenerationOptions though, and that's no longer happening. If it doesn't get set on the RazorCodeGenerationOptions, CI will continue to fail because compiler code paths that respond to this option will always take the false branch.

Reply:
Try FAR on `RazorCodeGenerationOptions.SuppressAddComponentParameter`. I'm pretty sure the suggested line above (which sets the option based on the RazorConfiguration) will make the source generator tests pass again. Also, you'll need to put back `SuppressAddComponentParameter` on `RazorCodeGenerationOptionsBuilder`.

To be clear, tooling _can't_ set this option for the Razor source generator.

Reply:
Good catch! Although, I don't think that will be enough. I think you'll find that the `RazorCodeGenerationOptionsBuilder` constructor that gets called in this case isn't the one that is passed a `RazorConfiguration`. FWIW, options in the compiler are... awkward and a bunch of refactoring is needed to make better. For example, there are two different "engine feature" interfaces for code generation option configuration and another two for parser option configuration. (It's on my list to clean that up.)

I still think you'll need to set the option here, but you can just grab it from the `razorSourceGeneratorOptions.Configuration.SuppressAddComponentParameter`. So, the option value would be coming from the configuration. At the very least, if that works and makes tests pass, we can clean it up when option configuration is cleaned up in the compiler.

Reply:
> I think you'll find that the RazorCodeGenerationOptionsBuilder constructor that gets called in this case isn't the one that is passed a RazorConfiguration 

Yea I just saw that :/